parser_test.go

```diff
@@ -45,6 +46,28 @@ func TestMultipleHost(t *testing.T) {
 	if ok := reflect.DeepEqual([]string{"google", "google2", "aws"}, h.Host); !ok {
 		t.Error("unexpected host mismatch")
 	}
+
+}
+
+// TestTrailingSpace ensures the parser does not hang when attempting to parse
+// a Host declaration with a trailing space after a pattern
+func TestTrailingSpace(t *testing.T) {
+	// in the config below, the first line is "Host google \n"
+	config := `
+Host googlespace 
+    HostName google.com
+`
+	sig := make(chan struct{})
+	go func() {
+		parse(config)
+		close(sig)
+	}()
+
+	select {
+	case <-sig:
+	case <-time.After(time.Millisecond * 50):
```

Comment:
I would drop the timeout from the test since this should not happen anymore. If it would happen the tests would timeout on the CI system.

Reply:
Okay, changed.  My reasoning was that the test just freezes `go test` which makes it difficult to ascertain what is going on if there is a regression.